les/fetcher.go

```diff
@@ -562,6 +562,10 @@ func (f *lightFetcher) checkAnnouncedHeaders(fp *fetcherPeerInfo, headers []*typ
 			// we ran out of recently delivered headers but have not reached a node known by this peer yet, continue matching
 			td = f.chain.GetTd(header.ParentHash, header.Number.Uint64()-1)
 			header = f.chain.GetHeader(header.ParentHash, header.Number.Uint64()-1)
+			if header == nil || td == nil {
+				log.Error("Missing parent of validated header", "hash", header.ParentHash, "number", header.Number.Uint64()-1)
```

Comment:
This will cause error if `header` is indeed `nil`

Reply:
Right, this one was a really stupid mistake.